File: java/src/main/java/ai/pagou/examples/payments/DemoData.java
package ai.pagou.examples.payments;

import ai.pagou.examples.lib.Models.Buyer;
import ai.pagou.examples.lib.Models.Document;
import ai.pagou.examples.lib.Models.ProductInput;
import java.util.List;

/** Synthetic buyer data — safe to commit. Never use real documents or PII. */
public final class DemoData {

  private DemoData() {}

  public static final Buyer DEMO_BUYER =
      new Buyer("Ana Souza", "ana.souza@example.com", new Document("CPF", "19100000000"));

  public static final List<ProductInput> DEMO_PRODUCTS = List.of(new ProductInput("Pro Plan", 4900, 1));
}

File: php/payments/DemoData.php
<?php

declare(strict_types=1);

namespace Pagou\Examples\Payments;

final class DemoData
{
    /**
     * Synthetic buyer data — safe to commit. Never use real documents or PII.
     * @return array<string, mixed>
     */
    public static function buyer(): array
    {
        return [
            'name' => 'Ana Souza',
            'email' => 'ana.souza@example.com',
            'document' => ['type' => 'CPF', 'number' => '19100000000'],
        ];
    }

    /** @return list<array<string, mixed>> */
    public static function products(): array
    {
        return [['name' => 'Pro Plan', 'price' => 4900, 'quantity' => 1]];
    }

    /** Reads a resource id from the first CLI argument or an env var. */
    public static function resourceIdFromArgs(string $envVar): string
    {
        $id = $GLOBALS['argv'][1] ?? getenv($envVar) ?: null;
        if (!$id) {
            throw new \RuntimeException("Pass a resource id as the first argument or set {$envVar}.");
        }
        return (string) $id;
    }
}
